### DISCUSSION TITLE
Warning Suppression

### DISCUSSION THREAD

**Discussion by scath999** [zephyrproject-rtos/zephyr#56827] [Q&A]
I'm getting a ton of warnings that aren't useful when building my app, which are hiding the really important issues.

This makes getting to the root of any problem a real hassle.

Can anyone provide/link-to guidance on how to suppress various warnings? I can't easily locate any.

Cheers.

**Comment by yashi** [comment]
Please provide a [minimal working example](https://en.wikipedia.org/wiki/Minimal_reproducible_example) with all the steps you are doing.  It might be depended on the compiler you are using.

To inhibit all warnings, do something like:

```
cmake_minimum_required(VERSION 3.20.0)

find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
project(my_project)

target_sources(app PRIVATE src/main.c)
target_compile_options(app PRIVATE -w)
```

But, Zephyr's warning setting is sane IMHO and any warnings should be fixed.

**Reply by aampdorian** [reply]
I am trying to find out how to suppress warnings also. Using VS code. I do not want to suppress all because I agree warnings should be fixed. However, sometimes we have callbacks that are called from pointers and get a [-Wunused-function] from the compiler. While I do care about most warnings, I want to ignore this one.

**Reply by AkselJensen** [reply]
Have you tried `target_compile_options(app PRIVATE -Wno-unused-function)` in your CMakeList.txt?
